File: app/src/main/java/ua/hospes/rtm/domain/team/TeamsRepository.java
package ua.hospes.rtm.domain.team;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import ua.hospes.rtm.domain.team.models.Team;

/**
 * @author Andrew Khloponin
 */
public interface TeamsRepository {
    /**
     * Get all teams from database
     */
    Observable<Team> get();

    Observable<Team> get(int id);

    /**
     * Get teams that not yet in Race
     */
    Observable<Team> getNotInRace();

    Observable<List<Team>> listen();

    Observable<Boolean> save(Team team);

    Single<Integer> remove(int id);

    Single<Integer> removeAll();
}

File: app/src/main/java/ua/hospes/rtm/domain/team/TeamsRepository.kt
package ua.hospes.rtm.domain.team

import kotlinx.coroutines.flow.Flow

internal interface TeamsRepository {
    suspend fun get(): List<Team>

    suspend fun get(vararg ids: Int): List<Team>

    suspend fun getNotInRace(): List<Team>


    fun listen(): Flow<List<Team>>


    suspend fun save(team: Team)

    suspend fun delete(id: Int)

    suspend fun clear()
}